src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -93,6 +94,8 @@ public override void ReadWrite(BitcoinStream stream)
             stream.ReadWrite(ref this.merkleProof);
             stream.ReadWrite(ref this.signature);
             stream.ReadWrite(ref this.coinstake);
+
+            this.HeaderSize = stream.Serializing ? stream.Counter.WrittenBytes : stream.Counter.ReadBytes;
```

Comment:
Is the the same we do in block size?

Reply:
Yes :

https://github.com/stratisproject/StratisBitcoinFullNode/blob/9f65b94274c82b93c5ce24ded759d0e84a7c237d/src/NBitcoin/Block.cs#L59

I wanted to keep the size of the header so `ProvenBlockHeaderStore` can determine if the `BatchThresholdSizeBytes` is reached :

https://github.com/ferdeen/StratisBitcoinFullNode/blob/b24ee608e6608531b28f6417992df07a8a3b3bcc/src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs#L54

Reply:
Also @fenix2222 has covered this in his PR - which will overwrite my implementation - which is good.

https://github.com/stratisproject/StratisBitcoinFullNode/blob/b11888465bebc4a85b56db98ace85ee28ab07bea/src/NBitcoin/ProvenBlockHeader.cs#L100-L113